packages/core/content-manager/admin/src/pages/EditView/components/FormInputs/Relations/RelationModal.tsx

```diff
@@ -71,16 +84,38 @@ const RelationModalWrapper = ({ relation, triggerButtonLabel }: RelationModalPro
   );
   const currentDocumentMeta = useDocumentContext('RelationModalBody', (state) => state.meta);
   const changeDocument = useDocumentContext('RelationModalBody', (state) => state.changeDocument);
+  const documentHistory = useDocumentContext('RelationModalBody', (state) => state.documentHistory);
+  const setDocumentHistory = useDocumentContext(
+    'RelationModalBody',
+    (state) => state.setDocumentHistory
+  );
 
   const [isConfirmationOpen, setIsConfirmationOpen] = React.useState(false);
-  const [isNavigating, setIsNavigating] = React.useState(false);
+  const [actionPosition, setActionPosition] = React.useState<'cancel' | 'back' | 'navigate'>(
+    'cancel'
+  );
 
   const [isModalOpen, setIsModalOpen] = React.useState(false);
+  // NOTE: Not sure about this relation modal context, maybe we should move this to DocumentContext?
   // Get parent modal context if it exists
   const parentContext = useRelationModal('RelationModalWrapper', (state) => state);
-  const isNested = parentContext.depth > 0;
-  // Track this modal's depth
-  const depth = isNested ? parentContext.depth + 1 : 1;
+  // Get depth of nested modals
+  const depth = parentContext ? parentContext.depth + 1 : 0;
+  // Check if this is a nested modal
+  const isNested = depth > 0;
+
+  const addDocumentToHistory = (document: DocumentMeta) =>
+    setDocumentHistory([...documentHistory, document]);
+
+  const getPreviousDocument = () => {
+    if (documentHistory.length === 0) return undefined;
+
+    const lastDocument = documentHistory[documentHistory.length - 1];
+
+    setDocumentHistory([...documentHistory].slice(0, documentHistory.length - 1));
```

Comment:
same think about the callback. But also why does a getter function set some state? This isn't what I would expect

Reply:
Good question, we should improve this

Reply:
I removed setDocumentHistory from the getter because it doesn't reflect what we want the getter to do, return the last element in history and I have created a single line function to remove the last item from history (I preffered to define a function instead of simply call setDocumentHistory([...documentHistory].slice(0, documentHistory.length - 1)); because the name of he function explains what we are trying to do)